earthly/cspell/Readme.md

```diff
@@ -0,0 +1,78 @@
+# CSpell Linter
+
+This Earthly Target and UDC enables uniform spell checking of all source and
+documentation files.
+
+## How it works
+
+Spellchecking is performed with the [`cspell`](cspell.org) program.
+
+Use the `CSPELL` Earthly UDC to enable uniform and consistent spell checking.
+
+This spellchecker is to be used in preference to tool specific spell checkers,
+such as `cargo spellcheck`.
+This is because we need to provide uniform and consistent spell checking across
+multiple source languages and documentation files.
+Tool specific spell checkers are limited in the kinds of files they can check,
+and also will use different configurations, dictionaries and project word lists.
+
+## Invocation
+
+In an Earthfile in your repo, add the following:
+
+```earthfile
+spellcheck-lint:
```

Comment:
Is there a particular reason we promote running it locally? I'm not sure I understand the justification at the end. The typical flow is copy source files into the container and then lint/check/test/build. In some cases, especially when doing code generation, the source files may change and be different than the ones stored locally.

Reply:
Because its running on Local code in the developers repo, or a fresh checkout in CI before any containers start getting built.  Its not running as part of a container being built.  And in fact the majority of time someone would want to run this the last thing they would want is to actually build a container.  They just want to know, do I have any markdown lint issues in the code I am working on.

Once its been run on the src, there is zero point in running it on code that's then been copied into a container.

Reply:
Understandable. I suppose I'm just gunshy to the idea of breaking isolation by using `LOCALLY`. The Earthly documentation also suggests caution when using this:

```
The LOCALLY command can be used in place of a FROM command, which will cause earthly to execute all commands under the target directly on the host system, rather than inside a container. Commands within a LOCALLY target will never be cached. This feature should be used with caution as locally run commands have no guarantee they will behave the same on different systems.
```

Both of those problems are valid: we're breaking isolation and caching at the same time - two things that Earthly works to guarantee against. 

Theoretically, I don't think there's a speed increase either. A user still has to wait for the Docker image to download to their machine before the command can execute. Versus pulling the source into a container and executing it there, which would still have a delay for the container to build,  but it would only happen once and then be cached. So both methods have some sort of first-time delay and the difference is probably negligible to really consider it a factor.

Reply:
@stevenj @jmgilman 
We should implement a spell checker that requires minimal engineering effort to run locally.